packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-filter.ts

```diff
@@ -0,0 +1,203 @@
+import {SearchEngine} from '../../../../app/search-engine/search-engine';
+import {
+  ConfigurationSection,
+  DateFacetSection,
+  SearchSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+import {configuration, dateFacetSet, search} from '../../../../app/reducers';
+import {determineFacetId} from '../../_common/facet-id-determinor';
+import {DateFacetValue} from '../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+import {deselectAllFacetValues} from '../../../../features/facets/facet-set/facet-set-actions';
+import {updateFacetOptions} from '../../../../features/facet-options/facet-options-actions';
+import {executeSearch} from '../../../../features/search/search-actions';
+import {
+  logFacetClearAll,
+  logFacetSelect,
+} from '../../../../features/facets/facet-set/facet-set-analytics-actions';
+import {
+  registerDateFacet,
+  RegisterDateFacetActionCreatorPayload,
+  updateDateFacetValues,
+} from '../../../../features/facets/range-facets/date-facet-set/date-facet-actions';
+import {validateDateFacetOptions} from './headless-date-facet-options';
+import {dateFacetResponseSelector} from '../../../../features/facets/range-facets/date-facet-set/date-facet-selectors';
+
+/**
+ * The options defining a `DateFilter`.
+ */
+export interface DateFilterOptions {
+  /**
+   * The field whose values you want to display in the filter.
+   */
+  field: string;
+
+  /**
+   * A unique identifier for the controller.
+   * By default, a unique random identifier is generated.
+   */
+  facetId?: string;
+
+  /**
+   * Whether to exclude folded result parents when estimating the result count for each facet value.
+   *
+   * @defaultValue `true`
+   */
+  filterFacetCount?: boolean;
+
+  /**
+   * The maximum number of results to scan in the index to ensure that the facet lists all potential facet values.
+   *
+   * Note: A high injectionDepth may negatively impact the facet request performance.
+   *
+   * Minimum: `0`
+   *
+   * @defaultValue `1000`
+   */
+  injectionDepth?: number;
+}
+
+export interface DateFilterRange {
+  /**
+   * The starting value for the date range.
+   */
+  start: string;
```

Comment:
I think we should explain the format of the string.

Another thing that would be nice is if we could reuse the DateFacet's `buildDateRange` utility which allows passing a number, or date object, providing more flexibility on how to configure the start and end.

Reply:
buildDateRange can be reused, only the start & end are required